### ISSUE TITLE
FESOM2 and ECHAM6 developments lack proper versioning

### ISSUE THREAD

**Issue by christian-stepanek** [esm-tools/esm_tools#75]
It is often necessary to trace down the exact version of a model that has been used for a simulation. With ECHAM6, version numbers are provided with each release. For own further developments of the model, and in particular for FESOM2, where no official version numbers are available, there is a lack of a simple procedure to define version numbers of submodels of AWI-CM/-ESM.

I propose to discuss how such versioning could be created via the esm-tools. Some ideas for discussion:

- When a simulation starts, the version of a model is written to a log file, potentially even a new log-file like "model_versions.info" that is dedicated to this specific purpose.

- For source codes that have not been changed w.r.t. the repository: Is there a way to automatically translate commit IDs to version numbers like 2.1.34? No idea whether and how this is possible. Yet, if it is possible, then that problem will be solved and the version number can be written to model_versions.info (or similar) upon simulation start.

- For source codes that have been changed w.r.t. the repository: When a simulation starts, is there any way to automatically compare the source code employed for building the executable w.r.t. to the repository version from which the source code stems and to list the differences in model_versions.info (or similar)?

It may take some time, brainstorming, and smart ideas to resolve this issue. But at a time, when lots of colleagues are working on model improvements in parallel, it would be very helpful to have an easy and practical way to derive the specific version, and list potential changes to the source code of the repository, from which a specific simulation stems.

**Comment by pgierz** [comment]
Some notes on how to do this:

1) Adding a new log file is trivial

2) We can get the tag version number from the commit like this: `git describe --exact-match HEAD || git rev-parse --short HEAD`, which will return either the tag name, or the commit ID if it is not a tagged commit.

3) Sure, `git diff @{push}` (See 3rd answer here: https://stackoverflow.com/questions/1800783/how-to-compare-a-local-git-branch-with-its-remote-branch). Needs a modern git though...

**Comment by pgierz** [comment]
Closing in favour of #99, these two problems should be merged.
